### PR TITLE
Fix duplication in the face of conditional returns

### PR DESCRIPTION
**Pull request by sb98052**
Release notes: Fixed code duplication bug

Issue: #1869 

The test has been confirmed manually for now. Resolving #1871

### CODE REVIEW

File: test/serializer/abstract/Return10.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+(function () {
+    x = global.__abstract?__abstract("number", "5"):5;
+    if (x==5) {
+	    console.log("42");
+    } else  {
+	    return;
+    }
+})()
+
+let inspect = () => logstream;
```
Comment (NTillmann) [inline test/serializer/abstract/Return10.js]: what is `logstream`?

File: test/serializer/abstract/Return10.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// count of 42:1
```
Comment (hermanventer) [inline test/serializer/abstract/Return10.js]: // Copies of 42:1

**Comment by hermanventer** [comment]
Thanks for doing this, it is very helpful for directing the search for a fix. At the moment, I think your fix treats the symptom and not the disease, so I'm not going to say go ahead with this, just yet.

**Comment by sb98052** [comment]
Closing. Superseded by #1875.

**Pull request closed without merging**